internal/c/libqb/src/threading-posix.cpp

```diff
@@ -0,0 +1,105 @@
+
+#include "libqb-common.h"
+
+#include <stdlib.h>
+#include <string.h>
+#include <pthread.h>
+
+#include "mutex.h"
+
+struct libqb_thread {
+    pthread_t thread;
+};
+
+struct libqb_mutex {
+    pthread_mutex_t mtx;
+};
+
+struct libqb_condvar {
+    pthread_cond_t var;
+};
+
+struct libqb_mutex *libqb_mutex_new() {
+    struct libqb_mutex *m = (struct libqb_mutex *)malloc(sizeof(*m));
+    pthread_mutex_init(&m->mtx, NULL);
+    return m;
+}
+
+void libqb_mutex_free(struct libqb_mutex *mutex) {
+    pthread_mutex_destroy(&mutex->mtx);
+    free(mutex);
+}
+
+void libqb_mutex_lock(struct libqb_mutex *m) {
+    if (m == NULL)
```

Comment:
This shouldn't be here, the Windows version doesn't have it and I don't think we should allow it to work this way anyway (silently not locking if the mutex is NULL)